app/src/main/java/pl/ludek/smat/home/pool_office_client/presentation/MainActivity.kt

```diff
@@ -22,51 +26,71 @@ class MainActivity : AppCompatActivity() {
     private lateinit var mainActivityViewModel: MainActivityViewModel
     private lateinit var binding: ActivityMainBinding
 
-
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         initView()
         initModel()
-
         setContentView(binding.root)
-
         updateData()
     }
 
     private fun initModel() {
-        mainActivityViewModel.poolInfoData.observe(this, Observer { poolInfoData ->
-                  binding.recyclerViewPoolInfo.layoutManager = LinearLayoutManager(this)
-                binding.recyclerViewPoolInfo.adapter = CustomAdapter(poolInfoData, poolInfoName)
+        mainActivityViewModel.poolInfoData.observe(this, Observer {networkResultPoolInfoData ->
+            when(networkResultPoolInfoData){
+                is NetworkResult.Success -> {
+                    val poolInfoData = networkResultPoolInfoData.bodyData as PoolInfoData
+                    binding.recyclerViewPoolInfo.layoutManager = LinearLayoutManager(this)
+                    binding.recyclerViewPoolInfo.adapter = CustomAdapter(poolInfoData, poolInfoName)
+                }
+                is NetworkResult.Error -> showToast(networkResultPoolInfoData.message.toString())
+                is NetworkResult.Exception -> showToast("Exception")
+            }
         })
-        mainActivityViewModel.completeRelayStateData.observe(this, Observer { completeRelayData ->
-            if (completeRelayData == null) {
-                showToast(errorRelayStr)
-            } else {
-                ignoreSwitchCheckedChange = true
-                completeRelayData.forEachIndexed { index, state ->
-                    if (index < binding.switchPane.size) {
-                        val switch = binding.switchPane[index] as SwitchCompat
-                        switch.isChecked = state
+
+        mainActivityViewModel.completeRelayStateData.observe(this, Observer { networkResultCompleteRelayData ->
+            when(networkResultCompleteRelayData){
+                is NetworkResult.Success -> {
+                    val completeRelayData = networkResultCompleteRelayData.bodyData as InitializationStateRelay
+                    if (completeRelayData == null) {
```

Comment:
In the [NetworkResponseCall:20](https://github.com/tomasz-ludek/pool-office-client/pull/12/files#diff-1f7265678882e798eceec5667e49dc471e20d389bee9a9856897d162229c7a2fR20) you perform the check and return `NetworkResult.Error` if `bodyData` is null. Therefore this check seems to be redundant and could most likely be removed.